neo/rawio/neuralynxrawio/neuralynxrawio.py

```diff
@@ -128,32 +131,58 @@ class NeuralynxRawIO(BaseRawIO):
         ("channel_id", "uint32"),
         ("sample_rate", "uint32"),
         ("nb_valid", "uint32"),
-        ("samples", "int16", (NcsSection._RECORD_SIZE)),
+        ("samples", "int16", NcsSection._RECORD_SIZE),
     ]
 
     def __init__(
-        self, dirname="", filename="", exclude_filename=None, keep_original_times=False, strict_gap_mode=True, **kargs
+            self,
+            dirname="",
```

Comment:
keep filename, but you can set it to None. 
same for exclude_filename. Keep it but set to None. If someone uses it we raise an error.

Reply:
Yes, since we're at it let's do the full refactor